### ISSUE TITLE
DDoS attack injection does not match the parameters

### ISSUE THREAD

**Issue by pepper-jk** [tklab-tud/ID2T#45]
**Issue by [carlos.garcia](https://git.tk.informatik.tu-darmstadt.de/carlos.garcia)**
_Monday May 14, 2018 at 11:46 GMT_
_Originally opened as https://git.tk.informatik.tu-darmstadt.de/SPIN/ID2T-toolkit/issues/122_

----

The number of packets injected does not match the user specified parameters. For example, the command
```
id2t -i mawi-2018-04-04/dump.pcap -a ddos inject.at-timestamp=1522825500 attack.duration=60 packets.per-second=10000
```

does not inject 10,000 packets in total in the range of 60 seconds.

**Comment by pepper-jk** [comment]
- [ ] DDoS:
  - [ ] pps * time (duration) = total packets generated
  - [ ] delay (between sending two packets): 0.08ms
  - [ ] max bandwidth
    - [ ]  separate by public and local!!!!!
    - [ ] param
    - [ ] defautl: estimate the bandwith from pcap
      - [ ] local: 100mbit (pcap > 100 -> 1gbit)
      - [ ] public: 10mbit (pcap > 10 -> +2mbit until upper border)
    - [ ] warning if dropping packets because of bandwidth
    - [ ] Warning if exceeding the attack duration
  - [ ] latency (replyrecieved-packetsent):
    - [ ] 3ms public
    - [ ] 0.9ms local

**Comment by pepper-jk** [comment]
Fixed with 93bc7214554dd8bbcb1635bde0ebc237013e89f4, 7ed1a906db0a2d64be8d06635987504ef3a16e8d and aacde93d3db4ac2f07d459d07403f823a1b11ab6, as well as every other related commit, e.g. 52286c546754ac0ddbbba7b04b822782d88598cb.
